ext/openssl/openssl.c

```diff
@@ -4465,11 +4465,13 @@ PHP_FUNCTION(openssl_pkey_get_private)
 	size_t passphrase_len = sizeof("")-1;
 	php_openssl_pkey_object *key_object;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z|s", &cert, &passphrase, &passphrase_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z|s!", &cert, &passphrase, &passphrase_len) == FAILURE) {
```

Comment:
Do "" and null have different meaning in this context?

Reply:
Semantically, I don't think so, but there are a few checks for `NULL` in the current code. I could convert `""` to `NULL` though, if that's ok. @bukka Do you have any insights?

Reply:
I think that "" will result in passing it to a callback that will compare the actual password with an empty string. Technically `""` should allow decrypting PEM protected by an empty password. Although not sure if that actually works like that. I guess I would have to try it. I'd say it's probably best to keep it separate and treat `""` and `NULL` differently.

Reply:
In reality it's really just an edge case because no one sane should protect password with an empty string... :)